node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -89,6 +98,51 @@ func NewOsDeviceConnectivityHelperScsiGeneric(executer executer.ExecuterInterfac
 	}
 }
 
+func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathdOutputByVolumeIds(volumeUids []string) (string, error) {
+	return r.Helper.GetMpathdOutputByVolumeId(volumeUids)
+}
+
+func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathDeviceName(volumePath string) (string, error) {
+	procMountsFileContent, err := ioutil.ReadFile(procMountsFilePath)
+	if err != nil {
+		return "", err
+	}
+	procMounts := string(procMountsFileContent)
+
+	return r.Helper.GetMpathDeviceNameFromProcMounts(procMounts, volumePath)
+}
+
+func (r OsDeviceConnectivityHelperScsiGeneric) GetMatchingVolumeIdToMpathd(mpathdOutput string,
+	mpathDeviceName string, volumeUids []string) (string, error) {
+
+	volumeId, _ := r.Helper.GetVolumeIdByMpathdName(mpathDeviceName, mpathdOutput)
+
+	dmPath := filepath.Join(devMapperPath, filepath.Base(strings.TrimSpace(mpathDeviceName)))
+	if volumeId != "" {
+		if r.Helper.IsMpathdMatchVolumeIdWithoutErrors(dmPath, volumeUids) {
+			return volumeId, nil
+		}
+	}
+
+	if err := r.Helper.ReloadMultipath(); err != nil {
+		return "", err
+	}
+
+	volumeId, err := r.Helper.GetVolumeIdByMpathdName(mpathDeviceName, mpathdOutput)
+	if err != nil {
+		return "", err
+	}
+
+	if volumeId == "" {
+		return "", &VolumeIdNotFoundForMultipathDeviceNameError{mpathDeviceName}
+	}
+
+	if _, err = r.Helper.IsMpathdMatchVolumeId(dmPath, volumeUids); err != nil {
```

Comment:
why make a function boolean if its returned bool value is ignored?
consider making it `AssertDmPathMatchesVolumeId(dmPath string, volumeIdVariations []string) (error)`

Reply:
done